### PR TITLE
Updated Monkeytype

### PR DESCRIPTION
**Pull request by Miodec**
Hi, i am the developer of Monkeytype and decided to add PreMiD support.

I noticed it was already started by @drull1000 so I expanded on it. I moved the original author to the contributor's list, as well as updated the clientid with his permission.

image

### CODE REVIEW

File: websites/M/Monkeytype/dist/metadata.json [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-    "typemonkey.com"
+    "monkeytype.com"
   ],
     "description": {
-    "en": "Monkeytype is a minimalistic typing test, featuring many test modes, an account and user configurable features."
+    "en": "A minimalistic, customisable typing website. Test yourself in various modes, track your progress and improve your typing speed."
   },
   "service": "Monkeytype",
-  "version": "1.0.1",
+  "version": "1.1.0",
```
Comment (ririxidev) [inline websites/M/Monkeytype/dist/metadata.json:21]: 2.0.0 kek

File: websites/M/Monkeytype/presence.ts [hunk 2]
```diff
@@ -1,37 +1,61 @@
...
+const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  },
+  path = document.location.pathname.toLowerCase();
+presenceData.startTimestamp = time;
+if (path === "/") {
+  presenceData.details = "Idling";
+  const resultPageVisible = !document.querySelector('.pageTest #result').classList.contains('hidden');
+  const testActive = document.querySelector('#words letter.correct') || document.querySelector('#words letter.incorrect');
+  const lbVisible = !document.querySelector('#leaderboardsWrapper').classList.contains('hidden');
```
Comment (ririxidev) [inline websites/M/Monkeytype/presence.ts:16]: merge consts

File: websites/M/Monkeytype/presence.ts [hunk 3]
```diff
@@ -1,37 +1,61 @@
...
+  if(resultPageVisible){
+    const testType = document.querySelector('.testType .bottom').innerHTML.replace(/<br>/g,' ');
+    presenceData.details = `Finished ${testType}`;
+    const statsElem = document.querySelector('.pageTest #result .stats');
+    const moreStatsElem = document.querySelector('.pageTest #result .morestats');
+    const stats = {
```
Comment (ririxidev) [inline websites/M/Monkeytype/presence.ts:22]: merge consts

File: websites/M/Monkeytype/presence.ts [hunk 4]
```diff
@@ -1,37 +1,61 @@
...
+    }
+    presenceData.state = `${stats.wpm} wpm ${stats.acc} acc ${stats.raw} raw ${stats.char} ${stats.con} consistency ${stats.time}`;
+  }else if(testActive){
+    const mode = document.querySelector('#top .config .mode .text-button.active').textContent.replace(/[\t\r\n]/g,'');
+    const testType = document.querySelector('.pageTest #premidTestMode').textContent;
+
+    presenceData.details = `Typing ${testType}`;
+
+    const wpm = document.querySelector('.pageTest #largeLiveWpmAndAcc #liveWpm').textContent;
+    const acc = document.querySelector('.pageTest #largeLiveWpmAndAcc #liveAcc').textContent;
```
Comment (ririxidev) [inline websites/M/Monkeytype/presence.ts:38]: merge consts?

**Review by ririxidev — CHANGES_REQUESTED** [review]
also can we get proof of your conv with formal owner? :)

**Comment by Miodec** [comment]
@ririxidev by merge consts im assuming you mean putting them in a single object? whats the advantage?

Heres the convo:
image

**Comment by ririxidev** [comment]
example of merging consts:
```javascript
const g = "gsgsgsg",
h = "sgsgsg",
jj = "fijswojsnmg";
```

**Comment by Miodec** [comment]
> example of merging consts:
> 
> ```js
> const g = "gsgsgsg",
> h = "sgsgsg",
> jj = "fijswojsnmg";
> ```

Right, makes sense.

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**